### DISCUSSION TITLE
Cant specialize "diagnostic.unnecessary" for variables / functions / members

### DISCUSSION THREAD

**Discussion by SneakyChocolate** [helix-editor/helix#11654] [Q&A] [answered]
I am configuring my own theme and i cant figure out how to make a "diagnostic.unnecessary" setting for different types of things such as variables and functions. everything has the same color but i know that themes such as "dark_plus" and "zed_onedark" have specific "unnecessary" configurations, so that functions and variable are darker than their original. Thanks for help in advance :)

**Comment by the-mikedavis** [comment, marked as answer]
There isn't a way to style unnecessary and deprecated diagnostics based on whether the diagnostic is on a function or variable: those diagnostics come from the language server while highlighting is done with tree-sitter. Those themes are using the `dim` modifier for `diagnostic.unnecessary` and `crossed_out` for `diagnostic.deprecated`. The default theme uses those modifiers too: https://github.com/helix-editor/helix/blob/6309cc71cc79106c74745f12bbdf1fccdabd98b6/theme.toml#L86-L87

Diagnostic highlights are overlaid onto syntax highlights so the effect is that you have regular variable/function/etc. highlighting with the extra modifiers
